spacy/cli/debug_data.py

```diff
@@ -861,6 +900,20 @@ def _compile_gold(
                 data["n_nonproj"] += 1
             if nonproj.contains_cycle(aligned_heads):
                 data["n_cycles"] += 1
+        if "trainable_lemmatizer" in factory_names:
+            # from EditTreeLemmatizer._labels_from_data
+            lemma_set = set()
+            for token in gold:
+                lemma_set.add(token.lemma)
```

Comment:
Seems more straightforward to add to the set after checking that `lemma != 0` and then check:

```
if len(lemma_set) == 0 and len(gold) > 1:
```

below?

Reply:
I agree that is more straightforward. We'd lose the debug case where every token in a document has the same (non-`0`) lemma, but I'm not sure how often that would actually happen in practice? I think the first part about not adding to the set if it is `0` is still a good improvement though.

Reply:
Ah right. Maybe the condition can be left as is (just moving the `add`), then both cases are covered.